### PR TITLE
[js] Added Node.js N-API bindings for Text2VideoPipeline

### LINKED ISSUE #3305: JS API for Video Generation
### Context

JavaScript bindings exist for `LLMPipeline` and `VLMPipeline` (see `src/js/src/addon.cpp`) but there is no JS wrapper for `Text2VideoPipeline`. The task is to add Node.js N-API bindings so that video generation can be used from JavaScript/TypeScript.

**What to do:**
- Create a C++ wrapper class (e.g. `Text2VideoPipelineWrapper`) under `src/js/src/` — follow the pattern in `src/js/src/llm_pipeline/llm_pipeline_wrapper.cpp`
- Register it in `src/js/src/addon.cpp` alongside the existing pipelines
- Add a TypeScript interface in `src/js/lib/pipelines/` — see `src/js/lib/pipelines/llmPipeline.ts` for reference
- Add tests and a usage sample

> **AI notice — Important!** We encourage contributors to use AI tools responsibly when creating Pull Requests. While AI can be a valuable aid, it is essential to ensure that your contributions meet the task requirements, build successfully, include relevant tests, and pass all linters. Submissions that do not meet these standards may be closed without warning to maintain the quality and integrity of the project. Please take the time to understand the changes you are proposing and their impact.

### PR DESCRIPTION
**Pull request by Nandkishore-04**
Fixes #3305

Added Node.js N-API bindings for `Text2VideoPipeline` following the existing VLM pipeline pattern. Includes C++ wrapper with  async init/generate, TypeScript types and factory method, tests, and a usage sample.

  ### Checklist:
  - [x] This PR follows GenAI Contributing guidelines.
  - [x] Tests have been updated or added to cover the new code.
  - [x] This PR fully addresses the ticket.
  - [x] I have made corresponding changes to the documentation.

### CODE REVIEW

File: samples/js/video_generation/text2video.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    console.log(`Generating video for prompt: "${prompt}"`);
+    const result = await pipe.generate(prompt, {
+        negative_prompt: 'worst quality, inconsistent motion, blurry, jittery, distorted',
+        height: 480,
+        width: 704,
+        num_frames: 161,
+        num_inference_steps: 25,
+        num_videos_per_prompt: 1,
+        guidance_scale: 3,
+        frame_rate: 25,
```
Comment (likholat) [inline samples/js/video_generation/text2video.js:28]: Could you please add a callback here, just a simple one that reports the current generation step?

**Comment by likholat** [comment]
@Retribution98 could you please take a look?

**Comment by Nandkishore-04** [comment]
hey @likholat , i have added the callback and fixed the ts error, the code is ready to review !

**Comment by Nandkishore-04** [comment]
hey @likholat @yatarkan, in the new commit i have Fixed TS type error in addon.ts, added step callback to text2video sample, resolved lint/formatting issues, and updated T2V test to skip gracefully when model path is not set.
it's ready for review! 😄

**Comment by Nandkishore-04** [comment]
hey @likholat @yatarkan , look like the 3 failing checks are unrelated to this PR — network timeouts in Python VLM tests, Trivy scanner installation failure in SDL tests, and the overall_status gate job failing as a result. let me know if there is any feedback on this. Thank you!

**Comment by likholat** [comment]
@Retribution98 could you please take a look?

File: src/js/src/text2video_pipeline/pipeline_wrapper.cpp [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+
+        if (config.negative_prompt.has_value()) {
+            config_obj.Set("negative_prompt", Napi::String::New(env, config.negative_prompt.value()));
+        }
+        if (config.guidance_rescale.has_value()) {
+            config_obj.Set("guidance_rescale", Napi::Number::New(env, config.guidance_rescale.value()));
+        }
+        if (config.frame_rate.has_value()) {
+            config_obj.Set("frame_rate", Napi::Number::New(env, config.frame_rate.value()));
+        }
```
Comment (Retribution98) [inline src/js/src/text2video_pipeline/pipeline_wrapper.cpp]: Please extract it into the cpp_to_js<ov::genai::VideoGenerationConfig, Napi::Value> helper.cpp function.

File: samples/js/video_generation/text2video.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
+import { Text2VideoPipeline } from 'openvino-genai-node';
+import { basename } from 'node:path';
+
+main();
```
Comment (Retribution98) [inline samples/js/video_generation/text2video.js:4]: Please add testing this sample in `tests/python_tests/samples/test_text2video.py`

File: src/js/lib/pipelines/text2VideoPipeline.ts [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+      properties["num_inference_steps"] = options.num_inference_steps;
+    if (options.guidance_scale !== undefined) properties["guidance_scale"] = options.guidance_scale;
+    if (options.frame_rate !== undefined) properties["frame_rate"] = options.frame_rate;
+    if (options.num_videos_per_prompt !== undefined)
+      properties["num_videos_per_prompt"] = options.num_videos_per_prompt;
+    if (options.max_sequence_length !== undefined)
+      properties["max_sequence_length"] = options.max_sequence_length;
+    if (options.guidance_rescale !== undefined)
+      properties["guidance_rescale"] = options.guidance_rescale;
+    if (options.callback !== undefined) properties["callback"] = options.callback;
```
Comment (Retribution98) [inline src/js/lib/pipelines/text2VideoPipeline.ts]: Let's remove this check and send all the options here. The native addon should convert the defined properties into a C++ entity.

File: src/js/lib/addon.ts [hunk 5]
```diff
@@ -172,11 +172,42 @@ export interface VLMPipeline {
   getGenerationConfig(): GenerationConfig;
 }
 
+export type VideoGenerationPerfMetrics = {
+  loadTime: number;
+  generateDuration: number;
+  iterationDuration: { mean: number; std: number };
+  transformerInferDuration: { mean: number; std: number };
+  vaeDecoderInferDuration: number;
+};
```
Comment (Retribution98) [inline src/js/lib/addon.ts]: Please move this entity to the `src/js/lib/perfMetrics.ts` and extend the list of properties.

File: src/js/lib/addon.ts [hunk 6]
```diff
@@ -172,11 +172,42 @@ export interface VLMPipeline {
...
+    properties: Record<string, unknown>,
+    callback: (err: Error | null) => void,
+  ): void;
+  generate(
+    prompt: string,
+    properties: Record<string, unknown>,
+    callback: (err: Error | null, result: Text2VideoResult) => void,
+  ): void;
+  getGenerationConfig(): Record<string, unknown>;
+  setGenerationConfig(config: Record<string, unknown>): void;
```
Comment (Retribution98) [inline src/js/lib/addon.ts]: Please use the strict type for both the GenerationConfig and the GenerationOptions.

File: src/js/lib/index.ts [hunk 7]
```diff
@@ -3,6 +3,7 @@
 
 import { LLMPipeline as LLM } from "./pipelines/llmPipeline.js";
 import { VLMPipeline as VLM } from "./pipelines/vlmPipeline.js";
+import { Text2VideoPipeline as T2V } from "./pipelines/text2VideoPipeline.js";
```
Comment (Retribution98) [inline src/js/lib/index.ts]: Please use the short name 'Text2Video'.

File: src/js/src/text2video_pipeline/pipeline_wrapper.cpp [hunk 8]
```diff
@@ -0,0 +1,253 @@
...
+                    auto trans_dur = perf.get_transformer_infer_duration();
+                    auto trans_obj = Napi::Object::New(env);
+                    trans_obj.Set("mean", Napi::Number::New(env, trans_dur.mean));
+                    trans_obj.Set("std", Napi::Number::New(env, trans_dur.std));
+                    perf_obj.Set("transformerInferDuration", trans_obj);
+
+                    perf_obj.Set("vaeDecoderInferDuration",
+                                 Napi::Number::New(env, perf.get_vae_decoder_infer_duration()));
+
+                    result_obj.Set("perfMetrics", perf_obj);
```
Comment (Retribution98) [inline src/js/src/text2video_pipeline/pipeline_wrapper.cpp]: Please extract it into the cpp_to_js<ov::genai::VideoGenerationPerfMetrics, Napi::Value> helper.cpp function.

File: src/js/src/text2video_pipeline/pipeline_wrapper.cpp [hunk 9]
```diff
@@ -0,0 +1,253 @@
...
+        *context->is_generating = false;
+        context->callback.Release();
+    };
+    try {
+        ov::genai::VideoGenerationResult result = context->pipe->generate(context->prompt, *context->properties);
+
+        napi_status status =
+            context->callback.BlockingCall([result, &report_error](Napi::Env env, Napi::Function jsCallback) {
+                try {
+                    auto result_obj = Napi::Object::New(env);
```
Comment (Retribution98) [inline src/js/src/text2video_pipeline/pipeline_wrapper.cpp]: Please extract it into the cpp_to_js<ov::genai::VideoGenerationResult, Napi::Value> helper.cpp function.

File: src/js/src/text2video_pipeline/pipeline_wrapper.cpp [hunk 10]
```diff
@@ -0,0 +1,253 @@
...
+        }
+        if (js_obj.Has("max_sequence_length") && !js_obj.Get("max_sequence_length").IsUndefined()) {
+            config.max_sequence_length = js_obj.Get("max_sequence_length").As<Napi::Number>().Int32Value();
+        }
+        if (js_obj.Has("guidance_rescale") && !js_obj.Get("guidance_rescale").IsUndefined()) {
+            config.guidance_rescale = js_obj.Get("guidance_rescale").As<Napi::Number>().FloatValue();
+        }
+        if (js_obj.Has("frame_rate") && !js_obj.Get("frame_rate").IsUndefined()) {
+            config.frame_rate = js_obj.Get("frame_rate").As<Napi::Number>().FloatValue();
+        }
```
Comment (Retribution98) [inline src/js/src/text2video_pipeline/pipeline_wrapper.cpp]: Please extract it into the js_to_cpp<ov::genai::VideoGenerationConfig> helper.cpp function. It should construct a new VideoGenerationConfig entity.

File: src/js/tests/text2VideoPipeline.test.js [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+
+import assert from "node:assert";
+import { describe, it, before } from "node:test";
+import { Text2VideoPipeline } from "../dist/index.js";
+
+const { T2V_PATH } = process.env;
+
+describe(
+  "Text2VideoPipeline",
+  { skip: !T2V_PATH ? "T2V_PATH environment variable is not set" : undefined },
```
Comment (Retribution98) [inline src/js/tests/text2VideoPipeline.test.js]: Please do not skip the tests if the T2V_PATH variable is not set. We won't have any details if it is not run.

Please use `optimum-intel-internal-testing/tiny-random-ltx-video` model for testing.
Add this model into the `src/js/tests/setup.py`.

File: src/js/tests/text2VideoPipeline.test.js [hunk 12]
```diff
@@ -0,0 +1,96 @@
...
+        num_inference_steps: 2,
+        height: 64,
+        width: 64,
+        num_frames: 9,
+      });
+
+      assert.ok(result, "Should return a result");
+      assert.ok(result.video, "Result should have a video tensor");
+    });
+  },
```
Comment (Retribution98) [inline src/js/tests/text2VideoPipeline.test.js]: Please add a test for the callback and for some corner cases if the parameters are invalid.

**Review by Retribution98 — COMMENTED** [review]
Thank you for your contributions. Please pay attention to my comments and make the necessary corrections.

Please update the documentation in the 'site' directory as well.

**Comment by Nandkishore-04** [comment]
hey @Retribution98 I've Addressed all review comments: extracted C++↔JS conversions into helper templates, moved types to proper locations, simplified generate(), added tests (callback + corner cases), updated site docs with JS tab. Also resolved merge conflicts with master.

File: src/js/src/text2video_pipeline/pipeline_wrapper.cpp [hunk 13]
```diff
@@ -0,0 +1,173 @@
...
+                        env.Null(),
+                        result_obj
+                    });
+                } catch (std::exception& err) {
+                    report_error("The final callback failed. Details:\n" + std::string(err.what()));
+                }
+            });
+
+        if (status != napi_ok) {
+            report_error("The final BlockingCall failed with status " + status);
```
[the remark this answers is not in the document]
Comment (Retribution98) [reply src/js/src/text2video_pipeline/pipeline_wrapper.cpp]: status won't be converted into readable status

**Comment by Retribution98** [comment]
@Nandkishore-04 please fix copilot comments and both Lint and SDL CI

**Comment by Nandkishore-04** [comment]
hey @Retribution98, I've fixed the copilot comments and remaining ci failures(SDL/lint). The code is ready for review!

**Comment by Nandkishore-04** [comment]
@Retribution98 implemented the copilots suggestion, code is ready for review!

**Comment by Nandkishore-04** [comment]
Hi @Retribution98 @likholat , I've addressed all the previous feedback (Copilot comments, lint/formatting fixes, site docs, merge conflicts). Would love to get your review when you have a chance. Happy to make any further changes if needed!

File: src/js/include/text2video_pipeline/pipeline_wrapper.hpp [hunk 14]
```diff
@@ -0,0 +1,27 @@
...
+
+    Napi::Value init(const Napi::CallbackInfo& info);
+    Napi::Value generate(const Napi::CallbackInfo& info);
+    Napi::Value get_generation_config(const Napi::CallbackInfo& info);
+    Napi::Value set_generation_config(const Napi::CallbackInfo& info);
+
+private:
+    std::shared_ptr<ov::genai::Text2VideoPipeline> pipe = nullptr;
+    std::shared_ptr<bool> is_initializing = std::make_shared<bool>(false);
+    std::shared_ptr<bool> is_generating = std::make_shared<bool>(false);
```
Comment (sgonorov) [inline src/js/include/text2video_pipeline/pipeline_wrapper.hpp:26]: Shouldn't we use `std::shared_ptr<std::atomic<bool>>` like in VLM here?

File: src/js/src/text2video_pipeline/pipeline_wrapper.cpp [hunk 15]
```diff
@@ -0,0 +1,171 @@
...
+    }
+    return env.Undefined();
+}
+
+Napi::Value Text2VideoPipelineWrapper::generate(const Napi::CallbackInfo& info) {
+    auto env = info.Env();
+    try {
+        OPENVINO_ASSERT(this->pipe, "Text2VideoPipeline is not initialized");
+        OPENVINO_ASSERT(!*this->is_generating, "Another generation is already in progress");
+        *this->is_generating = true;
```
Comment (sgonorov) [inline src/js/src/text2video_pipeline/pipeline_wrapper.cpp:113]: It needs to be moved down - if validate args count throws, state is already changed.

File: src/js/lib/pipelines/text2VideoPipeline.ts [hunk 16]
```diff
@@ -0,0 +1,87 @@
...
+    return this.pipeline.getGenerationConfig();
+  }
+
+  /**
+   * Set video generation configuration.
+   * @param config - Generation configuration parameters to set.
+   */
+  setGenerationConfig(config: VideoGenerationConfig): void {
+    if (!this.pipeline) throw new Error("Text2VideoPipeline is not initialized");
+    this.pipeline.setGenerationConfig(config);
```
Comment (sgonorov) [inline src/js/lib/pipelines/text2VideoPipeline.ts:78]: Maybe we should merge config here? Otherwise all optional fields will be set as default ones.

File: src/js/include/text2video_pipeline/init_worker.hpp [hunk 17]
```diff
@@ -0,0 +1,32 @@
+// Copyright (C) 2018-2026 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <napi.h>
+
+#include "openvino/genai/video_generation/text2video_pipeline.hpp"
+
+using namespace Napi;
```
Comment (sgonorov) [inline src/js/include/text2video_pipeline/init_worker.hpp:10]: Use Napi explicitly. 
```suggestion
```

File: src/js/src/text2video_pipeline/pipeline_wrapper.cpp [hunk 18]
```diff
@@ -0,0 +1,171 @@
...
+        report_error(e.what());
+        finalize();
+    }
+}
+
+Text2VideoPipelineWrapper::Text2VideoPipelineWrapper(const Napi::CallbackInfo& info)
+    : Napi::ObjectWrap<Text2VideoPipelineWrapper>(info) {};
+
+Napi::Function Text2VideoPipelineWrapper::get_class(Napi::Env env) {
+    return DefineClass(env,
```
Comment (sgonorov) [inline src/js/src/text2video_pipeline/pipeline_wrapper.cpp:79]: Need more exports. Like in VLM pipeline (reshape, decode, compile, clone...)

**Review by sgonorov — CHANGES_REQUESTED** [review]

**Comment by sgonorov** [comment]
Also please rebase

**Comment by Nandkishore-04** [comment]
hey, I'm gonna work on this!
